hack/update-codegen.sh

```diff
@@ -24,24 +24,28 @@ CODEGEN_PKG=$($GO_CMD list -m -f "{{.Dir}}" k8s.io/code-generator)
 
 cd $(dirname ${BASH_SOURCE[0]})/..
 
-chmod +x "${CODEGEN_PKG}/generate-internal-groups.sh"
-chmod +x "${CODEGEN_PKG}/generate-groups.sh"
-"${CODEGEN_PKG}/generate-groups.sh" \
-  applyconfiguration,client,lister,informer \
-  sigs.k8s.io/kueue/client-go \
-  sigs.k8s.io/kueue/apis \
-  kueue:v1beta1 \
-  --go-header-file ${KUEUE_ROOT}/hack/boilerplate.go.txt
+chmod +x "${CODEGEN_PKG}/kube_codegen.sh"
+source "${CODEGEN_PKG}/kube_codegen.sh"
+
+kube::codegen::gen_helpers \
+  --input-pkg-root sigs.k8s.io/kueue/apis \
+  --output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
+  --boilerplate ${KUEUE_ROOT}/hack/boilerplate.go.txt
+
+kube::codegen::gen_client \
+  --with-watch \
+  --input-pkg-root sigs.k8s.io/kueue/apis \
+  --output-pkg-root sigs.k8s.io/kueue/client-go --output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
+  --boilerplate ${KUEUE_ROOT}/hack/boilerplate.go.txt \
+  --with-applyconfig
 
 
 # Future releases of of code-generator add better support for out of GOPATH generation,
 # check https://github.com/kubernetes/code-generator/blob/master/examples/hack/update-codegen.sh for details,
 # for now we should just move the generated code to `client-go`.
-if  [ -d sigs.k8s.io/kueue/client-go ]
-then
```

Comment:
I would suggest avoiding unnecessary changes in this section, but would be happy to delete it entirely if possible now :). @trasc do you think this is still needed?

Reply:
I wold expect (and hope) that we can remove it all.

All the generators should support `--output-base`  now